frontend/views/containers/contributions/Contribution.vue

```diff
@@ -143,24 +150,34 @@ export default {
       return this.isFilled ? this.handleSubmit() : this.handleDelete()
     },
     handleDelete () {
-      this.$emit('new-value', 'nonMonetaryRemove', this.initialValue)
-      this.cancel()
+      if (this.ephemeral.isRemoving) { return }
+      this.ephemeral.isRemoving = true
+
+      this.$emit('new-value', 'nonMonetaryRemove', this.initialValue, () => {
+        this.ephemeral.isRemoving = false
+        this.cancel()
+      })
     },
-    handleSubmit () {
+    async handleSubmit () {
       if (this.$v.form.$invalid) {
         this.cancel()
       } else {
         if (this.isAdding) {
-          this.$emit('new-value', 'nonMonetaryAdd', this.form.contribution)
+          console.log('new-value called!', this.contributionsList)
+          await this.$listeners['new-value']('nonMonetaryAdd', this.form.contribution)
```

Comment:
Think some of these changes can be reverted now to use the `$emit` as before, see [comment below](https://github.com/okTurtles/group-income-simple/pull/854/files#r388661801)

Reply:
Please see the latest changes and comments.

Reply:
Hmm, what are you referring to specifically? Are you saying this way is the way it needs to be done, perhaps because of the `await`? (Did you double-check that's still true?)

Reply:
Yes, this way is the way it needs to be done. So we wait for the message to be processed before the loading state gets changed to `false` again.